### PR TITLE
formatList util

### LINKED ISSUE #1244: bad joining of strings on confirmation page
## Summary

Need better join on What could be affected section of Confirmation page.
image

Note that the How did the crime start? section has a join working correctly. We should extract this to a util function and use it in both places.

## Steps to reproduce

- Go to What could be affected page
- choose a bunch of checkboxes
- click through to confirmation page

## Unresolved questions

none

### PR DESCRIPTION
**Pull request by amazingphilippe**
Fixes #1244 

# Description

Added a utility to join strings together.

Need for launch because react-intl breaks Internet Explorer

# Checklist:

- [x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [x] I have added and needed tests for my changes (in particular for new screens)
- [x] I have added a comment to any confusing code

# Package changes

- Removed `react-intl`

# Required followup work

- Capitalization prevents items to be read as a sentence
- Need to integrate "other" as its custom input value
- figure out if we want default conjunction and glue translations. Or if we want them to be local to every `<DescriptionListItem/>` instance

### CODE REVIEW

File: f2/src/ConfirmationSummary.js [hunk 1]
```diff
@@ -94,20 +92,18 @@ export const ConfirmationSummary = () => {
       <Stack spacing={12} shouldWrapChildren>
         <HowDidItStartSummary />
         <WhatWasAffectedSummary />
-        {console.log('affectioned options: ' + impact.affectedOptions)}
```
Comment (amazingphilippe) [inline f2/src/ConfirmationSummary.js]: Remove this log 🌳

File: f2/src/locales/en.json [hunk 2]
```diff
@@ -86,6 +86,8 @@
   "contactinfoPage.skipButton": "Skip",
   "contactinfoPage.skipInfo": "Skip this step to report anonymously.",
   "contactinfoPage.title": "Your contact details",
+  "default.conjunction": ", and ",
```
Comment (sastels) [inline f2/src/locales/en.json]: as discussed IRL we need another conjunction for the case where there are two items.

File: f2/src/summary/HowDidItStartSummary.js [hunk 3]
```diff
@@ -31,23 +30,14 @@ export const HowDidItStartSummary = props => {
...
-      //compose the overview summary
-      overviewLine =
-        i18n._('confirmationPage.howDidItStart.overviewPrefix') +
-        summaryFirstitems +
-        i18n._('confirmationPage.howDidItStart.conjuction') +
-        summaryLastItem
-    }
+    overviewLine =
+      i18n._('confirmationPage.howDidItStart.overviewPrefix') +
+      formatList(summary, {
```
Comment (khalidelaggan) [inline f2/src/summary/HowDidItStartSummary.js]: Nice work!

File: f2/src/utils/formatList.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  if (size >= 3) {
+    let last = list.pop()
+    result = list.join(templateMiddle) + templateEnd + last
+  }
+
+  return result
+}
+
+//Make a component that extends <Text/> and joins parts of the list
```
Comment (khalidelaggan) [inline f2/src/utils/formatList.js]: +1

**Comment by sastels** [comment]
Looks great!

File: f2/src/summary/MoneyLostInfoSummary.js [hunk 5]
```diff
@@ -8,15 +8,31 @@ import { testdata, EditButton } from '../ConfirmationSummary'
...
+  moneyLost.methodPayment.map(key =>
+    methodPaymentSummary.push(
+      key === 'methodPayment.other' ? moneyLost.methodOther : i18n._(key),
+    ),
+  )
+  methodPaymentLine = formatList(methodPaymentSummary, {
+    pair: i18n._('default.pair'),
+    middle: i18n._('default.middle'),
+    end: i18n._('default.end'),
+  })
```
Comment (sastels) [inline f2/src/summary/MoneyLostInfoSummary.js]: 👍 I noticed this but figured we could do it as another PR rather than making you do it now 😁

**Pull request merged**